microsoft/electionguard-python
# 💻 CLI V1

[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
Fixes #561 

### Description
Adds a initial CLI program into election guard.  Thanks to [click](https://click.palletsprojects.com/en/8.1.x/) it contains command line documentation and the potential to have multiple commands.  

The code in this PR has two commands 1. a simple hello world to demonstrate how a second command will look.  2. an `e2e` command.

The e2e command will allow the running of an election including key ceremony, election building, submitting, encrypting, spoiling and casting votes, and tally and spoiled ballot decryption.  It does not yet publish an election record.  It currently takes two arguments: number of guardians and quorum.  It will eventually take a manifest and ballots, but not yet for this PR.

### Testing

1. Install the CLI per the readme.md document
2. Try `eg --help`

Expected: A list of the commands available

3. `eg e2e --help`

Expected: A description of the e2e command's parameters and usage

4. `eg e2e`

Expected: It should prompt you for missing parameters like quorum and number of guardians

5. `eg e2e --guardian-count=2 --quorum=2`

It should run through an election and print out results
